app/models/item.rb

```diff
@@ -1,3 +1,13 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def supermarket_name
+    supermarket.name
```

Comment:
Nice work here! It's also okay to call `item.supermarket.name` in the view since you are just grabbing the attributes from the association (rather than doing an AR calcuation)